tools/cellxgene_census_builder/src/cellxgene_census_builder/data_copy.py

```diff
@@ -0,0 +1,79 @@
+import argparse
+import logging
+import pathlib
+import subprocess
+import sys
+from typing import Union
+
+from .logging import logging_init_params
+
+
+def sync_to_S3(from_path: Union[str, pathlib.PosixPath], to_path: str, dryrun: bool = False) -> None:
+    """Copy (sync) local directory to S3.
+
+    Equivalent of `aws s3 sync local_directory_path S3_path`.
+
+    Raises upon error.
+    """
+    from_path = pathlib.PosixPath(from_path).absolute()
+    if not from_path.is_dir():
+        raise ValueError(f"Local path is not a directory: {from_path.as_posix()}")
+    if not to_path.startswith("s3://"):
+        raise ValueError(f"S3_path argument does not appear to be an S3 path: {to_path}")
+
+    cmd = ["aws", "s3", "sync", from_path.as_posix(), to_path, "--no-progress"]
+    if dryrun:
+        cmd += ["--dryrun"]
+
+    returncode = -1
+    try:
+        _log_it(f"Starting copy {from_path.as_posix()} -> {to_path}", dryrun)
+        with subprocess.Popen(cmd, stderr=subprocess.STDOUT, stdout=subprocess.PIPE, text=True) as proc:
+            if proc.stdout is not None:
+                for line in proc.stdout:
+                    logging.info(line)
+
+        returncode = proc.returncode
+        if returncode:
+            raise subprocess.CalledProcessError(returncode, proc.args)
+
+    finally:
+        _log_it(f"Completed copy, return code {returncode}, {from_path.as_posix()} -> {to_path}", dryrun)
+
+
+def _log_it(msg: str, dryrun: bool) -> None:
+    logging.info(f"{'(dryrun) ' if dryrun else ''}{msg}")
+
+
+def main() -> int:
+    description = """Sync (copy) a local directory to an S3 location."""
+    epilog = """Example:
+
+    python -m cellxgene_census_builder.data_copy /tmp/data/ s3://bucket/path/ --dryrun
+    """
+    parser = argparse.ArgumentParser(
+        prog="cellxgene_census_builder.data_copy",
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+        description=description,
+        epilog=epilog,
+    )
+    parser.add_argument("from_path", type=str, help="Data source, specified as a local path, e.g., /home/me/files/")
+    parser.add_argument("to_path", type=str, help="S3 path data is copied to, e.g., s3://bucket/path/")
+    parser.add_argument(
+        "--dryrun",
+        action=argparse.BooleanOptionalAction,
+        default=True,
+        help="Display, but do NOT perform actions. Useful for previewing actions. Default: True",
```

Comment:
```suggestion
        help="Display, but do NOT perform S3 data update actions, including Census data upload, logs upload, and removal of older Census releases. Useful for previewing actions. Default: True",
```

Reply:
This suggested edit does not make any sense in this test __main__.  The only function provided by this CLI is to sync/copy a user-specified directory to S3 (any directory).  It is completely agnostic about which directory and which S3 path.

Can you clarify what you are trying to accomplish?

Reply:
Agreed, sorry, wrong place for the edit, this is not top-level entryp oint.  If there's a better place to doc what dryrun does overall (config file), that's where I'd put it.

Reply:
This is minor, don't give it too much thought. :).  I simply figured it would be good to document in one place what dryrun does in total, since  it affects a few of the pipeline's steps. "Skips all S3 updates" is sufficient.

Reply:
Also, the config docs in `build_state.py` already contained:
> "dryrun": False,   # if True, will disable copy of data/logs/reports to S3 buckets

Reply:
Expanded comment, and extended this config to apply to ALL S3 writes (including the write of release.json).